django/contrib/gis/db/models/functions.py

```diff
@@ -372,15 +373,15 @@ class Perimeter(DistanceResultMixin, OracleToleranceMixin, GeoFunc):
     def as_postgresql(self, compiler, connection):
         function = None
         if self.geo_field.geodetic(connection) and not self.source_is_geography():
-            raise NotImplementedError("ST_Perimeter cannot use a non-projected non-geography field.")
+            raise NotSupportedError("ST_Perimeter cannot use a non-projected non-geography field.")
         dim = min(f.dim for f in self.get_source_fields())
         if dim > 2:
             function = connection.ops.perimeter3d
         return super().as_sql(compiler, connection, function=function)
 
     def as_sqlite(self, compiler, connection):
         if self.geo_field.geodetic(connection):
-            raise NotImplementedError("Perimeter cannot use a non-projected field.")
+            raise NotSupportedError("Perimeter cannot use a non-projected field.")
```

Comment:
This makes me wonder if this shouldn't be moved to `check_expression_support` instead?